### PR TITLE
Add RSDP support detection to u-root and adapt boot command to pass acpi_rsdp to the new kernel

### PR DESCRIPTION
**Pull request by vejmarie**

### CODE REVIEW

File: cmds/boot/boot.go [hunk 1]
```diff
@@ -363,6 +364,33 @@ func kexecLoad(grubConfPath string, grub []string, mountPoint string) error {
 	}
 	// defer ramfs.Close()
 
+	// if /tmp/rsdp file exist we must get the value and add it to the be.cmdline parameter
+	// this will allow the kernel to properly read the ACPI table
+	if _, err := os.Stat("/tmp/rsdp"); err == nil {
```
Comment (rminnich) [inline cmds/boot/boot.go]: this seems like too much work to me.
b, err := ioutil.ReadFile("/tmp/extracmdline")
if err != nil {
you can check for err.IsNotExist() or whatever it is
return
}

be.cmdline = append(be.cmdline, strings.Split(string(b), "\n"))

File: cmds/rsdp/rsdp.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	return returnValue, err
+}
+
+func main() {
+	rsdp_value, _ := getRSDP("/dev/kmsg")
+	f, err := os.Create("/tmp/rsdp")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
```
Comment (rminnich) [inline cmds/rsdp/rsdp.go]: f.Fprintf(f, "rsdp=%08x", rsdp_value)

File: cmds/boot/boot.go [hunk 3]
```diff
@@ -363,6 +364,33 @@ func kexecLoad(grubConfPath string, grub []string, mountPoint string) error {
...
+		var rsdp string
+		rsdp = " acpi_rsdp="
+		rsdpDesc, err_open := os.OpenFile("/tmp/rsdp", os.O_RDONLY, 0)
```
Comment (bkochendorfer) [inline cmds/boot/boot.go]: I think you want to use camel case instead of snake case here `errOpen`

File: cmds/boot/boot.go [hunk 4]
```diff
@@ -363,6 +364,33 @@ func kexecLoad(grubConfPath string, grub []string, mountPoint string) error {
...
+		if err_open != nil {
+			verbose("%v", err)
+			return err_open
+		}
+		r := bufio.NewReader(rsdpDesc)
+		address, err_read := r.ReadString('\n')
```
Comment (bkochendorfer) [inline cmds/boot/boot.go]: Same here re: casing

File: cmds/boot/boot.go [hunk 5]
```diff
@@ -363,6 +364,33 @@ func kexecLoad(grubConfPath string, grub []string, mountPoint string) error {
...
+		if err_read != nil {
+			if err_read != io.EOF  {
+				verbose("%v", err)
+				return err_read
+			}
+		}
+		rsdp = rsdp + address
+		log.Printf("Kernel ACPI parameters %s",rsdp)
```
Comment (bkochendorfer) [inline cmds/boot/boot.go]: Need a space, `log.Printf("Kernel ACPI parameters %s", rsdp)`.

File: cmds/boot/boot.go [hunk 6]
```diff
@@ -363,6 +364,33 @@ func kexecLoad(grubConfPath string, grub []string, mountPoint string) error {
...
+		be.cmdline = be.cmdline + rsdp
+		log.Printf("Kernel cmdline %s",be.cmdline)
```
Comment (bkochendorfer) [inline cmds/boot/boot.go]: Same as above, might need a formatter on this file.

File: cmds/rsdp/rsdp.go [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+			mystring := scanner.Text()
+			channel <- mystring
+		}
+		close(channel)
+		if err := scanner.Err(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	var data_read int
```
Comment (bkochendorfer) [inline cmds/rsdp/rsdp.go]: `dataRead`

File: cmds/rsdp/rsdp.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	"time"
+)
+
+func getRSDP(path string) (string, error) {
+
+	file, err := os.Open(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
```
Comment (bkochendorfer) [inline cmds/rsdp/rsdp.go]: I'd check the error and return / log

**Comment by vejmarie** [comment]
Just tried to take into account all of your positive remarks ;)

File: cmds/boot/boot.go [hunk 9]
```diff
@@ -363,6 +363,16 @@ func kexecLoad(grubConfPath string, grub []string, mountPoint string) error {
 	}
 	// defer ramfs.Close()
 
+	// if /tmp/rsdp file exist we must get the value and add it to the be.cmdline parameter
+	// this will allow the kernel to properly read the ACPI table
+
+	rsdp, err := ioutil.ReadFile("/tmp/rsdp")
+	if err == nil {
```
Comment (rminnich) [inline cmds/boot/boot.go]: do you need to also add a space between the two strings? I would expect it's a good caution.

File: cmds/rsdp/rsdp.go (hunk 2 above)
Comment (rminnich) [reply cmds/rsdp/rsdp.go]: ah got it you added a space pre emptively :_

**Comment by rminnich** [comment]
This is too useful. I'm pushing it. Jean-marie is kinda busy.

**Pull request merged**